volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -43,104 +69,233 @@ def _generator(self):
-
                 while attr.Attr_Header.AttrType.is_valid_choice:
-                    vollog.debug(f"Attr Type: {attr.Attr_Header.AttrType.lookup()}")
-
-                    # MFT Flags determine the file type or dir
-                    # If we don't have a valid enum, coerce to hex so we can keep the record
-                    try:
-                        mft_flag = mft_record.Flags.lookup()
-                    except ValueError:
-                        mft_flag = hex(mft_record.Flags)
-
-                    # Standard Information Attribute
-                    if attr.Attr_Header.AttrType.lookup() == "STANDARD_INFORMATION":
-                        attr_data = attr.Attr_Data.cast(si_object)
-                        yield 0, (
-                            format_hints.Hex(attr_data.vol.offset),
-                            mft_record.get_signature(),
-                            mft_record.RecordNumber,
-                            mft_record.LinkCount,
-                            mft_flag,
-                            renderers.NotApplicableValue(),
-                            attr.Attr_Header.AttrType.lookup(),
-                            conversion.wintime_to_datetime(attr_data.CreationTime),
-                            conversion.wintime_to_datetime(attr_data.ModifiedTime),
-                            conversion.wintime_to_datetime(attr_data.UpdatedTime),
-                            conversion.wintime_to_datetime(attr_data.AccessedTime),
-                            renderers.NotApplicableValue(),
-                        )
-
-                    # File Name Attribute
-                    if attr.Attr_Header.AttrType.lookup() == "FILE_NAME":
-                        attr_data = attr.Attr_Data.cast(fn_object)
-                        file_name = attr_data.get_full_name()
-
-                        # If we don't have a valid enum, coerce to hex so we can keep the record
-                        try:
-                            permissions = attr_data.Flags.lookup()
-                        except ValueError:
-                            permissions = hex(attr_data.Flags)
-
-                        yield 1, (
-                            format_hints.Hex(attr_data.vol.offset),
-                            mft_record.get_signature(),
-                            mft_record.RecordNumber,
-                            mft_record.LinkCount,
-                            mft_flag,
-                            permissions,
-                            attr.Attr_Header.AttrType.lookup(),
-                            conversion.wintime_to_datetime(attr_data.CreationTime),
-                            conversion.wintime_to_datetime(attr_data.ModifiedTime),
-                            conversion.wintime_to_datetime(attr_data.UpdatedTime),
-                            conversion.wintime_to_datetime(attr_data.AccessedTime),
-                            file_name,
-                        )
+                    for record in attr_callback(
```

Comment:
Looks like you could *probably* use `yield from` here, but honestly this is fine.  More just a point of interest...
https://docs.python.org/3/whatsnew/3.3.html#pep-380